subtree/rabit/src/allreduce_base.cc

```diff
@@ -15,6 +15,11 @@
 
 namespace rabit {
 namespace engine {
+
+namespace {
+const int kConnectRetryTimes = 5;
```

Comment:
Consider make this configurable, like other parameters, via DMLC_WORKER_CONNECT_RETRY